cargo/.github/workflows/rust_ci.yml

```diff
@@ -0,0 +1,51 @@
+name: Continuous Integration
+
+on:
+  push:
+    paths-ignore:
+      - "**/README.md"
+  pull_request:
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  {% raw -%}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  {%- endraw %}
+
+jobs:
+  rust-checks:
+    name: Rust Checks
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        action:
+          - command: build
+            args: --release
+          - command: fmt
+            args: --all -- --check --color always
+          - command: clippy
+            args: --all-targets --all-features --workspace -- -D warnings
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Enable caching
+        uses: Swatinem/rust-cache@v2
+      - name: Setup Rust
+      {%- if arch == "riscv" %}
+        uses: dtolnay/rust-toolchain@v1
+        with:
+          toolchain: nightly
+          components: rust-src
+      {%- else %}
+        uses: esp-rs/xtensa-toolchain@v1.5
+        with:
+          default: true
+          buildtargets: {{ mcu }}
+          ldproxy: false
+      {%- endif %}
+      - name: Run command
```

Comment:
What about having `cargo fmt` and maybe even `cargo clippy` in the CI by default?

Reply:
We currently check `build`, `fmt` and `clippy`